src/AnnotatedCommand.php

```diff
@@ -225,8 +225,9 @@ protected function setCommandArguments($commandInfo)
      */
     protected function checkUsesInputInterface($params)
     {
+        /** @var \ReflectionParameter $firstParam */
         $firstParam = reset($params);
-        return $firstParam instanceof InputInterface;
+        return $firstParam->getClass() && $firstParam->getClass()->implementsInterface(InputInterface::class);
```

Comment:
Thanks for `is_object()`, added it.

I like `instanceof`, but it just doesn't work here. That's the initial issue.

`$firstParam` is an instance of `ReflectionParameter` and `$firstParam->getClass()` is an instance of `ReflectionClass`. If you use `instanceof` with them, you check whether `ReflectionParameter` or `ReflectionClass` implements `InputInterface`. Not the original parameter type hint class.